src/Core/Style.js

```diff
@@ -160,39 +281,39 @@ class Style {
             this.point.radius = readVectorProperty(layer.paint['circle-radius'], zoom) || 1.5;
         } else if (layer.type === 'symbol') {
             // overlapping order
-            this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-z-order'] || 'auto', zoom);
+            this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-z-order'], zoom) || 'auto';
             if (this.text.zOrder == 'auto') {
-                this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-sort-key'] || 'Y', zoom);
+                this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-sort-key'], zoom) || 'Y';
             } else if (this.text.zOrder == 'viewport-y') {
                 this.text.zOrder = 'Y';
             } else if (this.text.zOrder == 'source') {
                 this.text.zOrder = 0;
             }
 
             // position
-            this.text.anchor = readVectorProperty(layer.layout && layer.layout['text-anchor'] || 'center', zoom);
-            this.text.offset = readVectorProperty(layer.layout && layer.layout['text-offset'] || [0, 0], zoom);
-            this.text.padding = readVectorProperty(layer.layout && layer.layout['text-padding'] || 2, zoom);
-            this.text.size = readVectorProperty(layer.layout && layer.layout['text-size'] || 16, zoom);
-            this.text.placement = readVectorProperty(layer.layout && layer.layout['symbol-placement'] || 'point', zoom);
-            this.text.rotation = readVectorProperty(layer.layout && layer.layout['text-rotation-alignment'] || 'auto', zoom);
+            this.text.anchor = readVectorProperty(layer.layout && layer.layout['text-anchor'], zoom) || 'center';
```

Comment:
Could you use the default value from constructor, instead of `'center'`.
Same comment for all properties.

May be write a new function:
```js
setPropertyFromLayout(this.text, 'offset', layer.layout, zoom);
```

Reply:
I changed them all to use the default value, otherwise no new function as it would be overkill: it checks `layer.layout` and `layer.paint` one time only